bindings/java/src/main/com/apple/foundationdb/tuple/TupleUtil.java

```diff
@@ -63,6 +64,28 @@
 	private static final byte[] TRUE_ARR           = new byte[]{0x27};
 	private static final byte[] VERSIONSTAMP_ARR   = new byte[]{0x33};
 
+	private static final ThreadLocal<DecodeResult> DECODE_RESULT_THREAD_LOCAL =
+			new ThreadLocal<DecodeResult>() {
+				@Override
+				protected DecodeResult initialValue() {
+					return new DecodeResult();
+				}
+			};
+	private static final ThreadLocal<EncodeResult> ENCODE_RESULT_THREAD_LOCAL =
```

Comment:
But in that case if we call Tuple.pack() n times it will create n EncoderResult objects. With ThreadLocal only 1 object for each thread.
Isn't ThreadLocal better in this case?

Reply:
In sense, yes, that's true in that n allocations is worse than 1 if we're using "number of allocations" as the metric. I'm not sure it follows that using a thread local is better if we're using "ability to maintain" as the metric (or a combination of ability to maintain and performance, which I think is probably the right metric).

Do you know what kind of performance boost making this thread local gets us (from some kind of numerical perspective)? I think the `EncodeResult` would be fairly short-lived if we allocate it once at the beginning of `Tuple` packing and then discard it at the end.

Reply:
Maybe I don't understand well what do you want to say, but I think in this case threadlocals will be better. It is possible that EncodeResult will live very short, but in any case during each pack() call it will allocate some memory and so the performance will be worse than using threadlocals. But I can't say any numbers, how faster it will be (it depends on situation, how many times the pack function is called). 
I don't think that threadlocals have any disadvantages in this case. What do you think?